### PR TITLE
Update Nomogram widget

### PR DESCRIPTION
**Pull request by kernc**
##### Issue
Fixes https://github.com/biolab/orange3/issues/2280

##### Description of changes
Fix slowness by only recomputing what is selected/drawn.
Also split main view into three parts.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by kernc** [comment]
Thanks! Hopefully resolved by going with this second approach?

File: Orange/widgets/visualize/tests/test_ownomogram.py [hunk 1]
```diff
@@ -23,8 +23,7 @@ def setUpClass(cls):
         cls.lenses = Table("lenses")
 
     def setUp(self):
-        self.widget = self.create_widget(OWNomogram)
-        self.widget.repaint = True
+        self.widget = self.create_widget(OWNomogram)  # type: OWNomogram
```
Comment (kernc) [inline Orange/widgets/visualize/tests/test_ownomogram.py]: The two length-asserting test failures can be mitigated by setting in `setUp()`:
```py
self.widget.display_index = 0  # i.e. display all attributes
```
Anybody has an idea of an alternative solution, and perhaps something for the third failing test as well?

**Comment by ajdapretnar** [comment]
TODO:  check split window since it results in duplication.

**Comment by ajdapretnar** [comment]
The view now seems better than at the last check, but if I can nitpick, it would benefit from an additional 3 mm of extra space above Total. :)

screen shot 2017-07-10 at 09 22 52

However, a more serious issue is that the visualization fails when trying to resize.

---------------------------- IndexError Exception -----------------------------
Traceback (most recent call last):
  File "/Users/ajda/orange/orange3/Orange/widgets/visualize/ownomogram.py", line 673, in resizeEvent
    self.w.update_scene()
  File "/Users/ajda/orange/orange3/Orange/widgets/visualize/ownomogram.py", line 878, in update_scene
    self.set_feature_marker_values(old_marker_values)
  File "/Users/ajda/orange/orange3/Orange/widgets/visualize/ownomogram.py", line 1069, in set_feature_marker_values
    item.dot.move_to_val(marker_values[i])
IndexError: list index out of range

**Comment by ajdapretnar** [comment]
Now, as far as I'm concerned, it's great!

**Pull request merged**